xmipp

```diff
@@ -702,10 +709,32 @@ def configCuda(configDict):
     if configDict["CUDA"]=="True":
         if configDict["NVCC"]=="":
             if checkProgram("nvcc"):
+                nvccVersion, nvccFullVersion = getCudaVersion("nvcc")
+                if nvccVersion != 8.0:
+                    print(red('Detected CUDA-' + nvccFullVersion + '. Version 8.0 is required.'))
+                    print(red("Skipping CUDA compilation. Please, set cuda-8.0 as default "
+                              "to be able to use Xmipp Cuda programs.'"))
+                    configDict["CUDA"] = 'False'
+                    return
                 configDict["NVCC"] = "nvcc"
         if configDict["NVCC_CXXFLAGS"]=="":
-            configDict["NVCC_CXXFLAGS"] = "--x cu -D_FORCE_INLINES -Xcompiler " \
-                                              "-fPIC -Wno-deprecated-gpu-targets -ccbin %s -std=c++11" % configDict["CXX"]
+            cxxVersion, cxxStrVersion = getGccVersion(configDict["CXX"])
+            if cxxVersion > 5:
+                print(red("%s version later than 5 detected. Finding alternative for CUDA compilations..." % configDict["CXX"]))
+                if checkProgram(configDict["CXX"]+"-5"):
+                    configDict["CXX_CUDA"] = configDict["CXX"]+"-5"
```

Comment:
This part is getting very long with many conditions.

How about we detect available Gcc versions, and set CXX and CXX_CUDA to version 4.8 or 5?

Reply:
CXX should be the as highest as possible because is not giving problems and we can take advantages in performance or whatever in the new versions. Though, if CXX is later than 5, CUDA-8.0 is not supporting it, so this fix that problem.

I don't know if I get your question, but I think that is not getting long, but flexible.

Reply:
What I meant is that, since GCC and Cuda are dependent, we should set both CXX and CXX_CUDA in one function (and maybe even to the same value, to avoid possible issues).

Ultimately, do we actually need this change for the release? 
I am asking because the whole xmipp script got to the point where a major refactoring would be desirable -> we can introduce these changes to the new version for the next release

Reply:
I agree in the cleaning and refactoring this script in the devel branch. We should clarify it and clean it in deep.

But yes, this is a need for the release because in OS where the default gcc/g++ is later than 5 (ubuntu18, openSUSE15), the CUDA compilation is failing.

I check this approach using gcc/g++ v8 for the nonCuda code and g++-5 for the cuda code (also the g++-4.8).